### ISSUE TITLE
Taken windows sometimes become stuck and inaccessible

### ISSUE THREAD

**Issue by cewood** [paperwm/PaperWM#895]
**Describe the bug**
Taken windows sometimes become stuck and inaccessible

**To Reproduce**
Steps to reproduce the behaviour:
1. Select/take some windows
2. Do nothing, or move to another desktop
3. Windows remain taken and are inaccessible
4. After locking the screen/computer, and unlocking, windows become untaken and accessible again

**Expected behavior**
The windows should be dropped/untaken, and go back to normal.

**Screenshots**
This screenshot is after the taken windows have become stuck. After this happens, the only way to recover the windows, is to lock and unlock the screen/computer.
![Screenshot from 2024-07-17 17-17-40](https://github.com/user-attachments/assets/fa7d0014-e0ee-4d7c-be57-f8089483ffb7)

**System information:**
```
Distribution: Arch Linux
GNOME Shell: 46.2
Display server: Wayland
PaperWM version: 46.10.1
Enabled extensions:
- system-monitor@gnome-shell-extensions.gcampax.github.com
- appindicatorsupport@rgcjonas.gmail.com
- paperwm@paperwm.github.com
- caffeine@patapon.info
```

**Comment by jtaala** [comment]
Thanks for reporting.  I've never seen this and haven't been able to reproduce this as yet.

First thing though, please update to the latest PaperWM.  `v46.10.1` is very old now and there's been much changes and some fixes since that release (which was over 4 months ago).

**Comment by jtaala** [comment]
Latest version is `v46.13.2` btw.

**Comment by jtaala** [comment]
From your screenshot, that looks like a very old version of PaperWM's "take windows" functionality.  Newer versions should have a floating "tooltip" in front of taken windows that describe new functionality... unless you're hitting some kind of issue that's stopping it from showing(?).

Tell me, does the floating tooltip sometimes show - you can see it in the videos in https://github.com/paperwm/PaperWM/pull/808.

**Comment by jtaala** [comment]
Update to the latest version and see if the issue remains - if so, I'll get you to do some digging (and looking in logs etc.).

Also, let me know if you're running anything that might interfere with keybinds (actually, let me know what keybind you're using for "take" as well).

**Comment by cewood** [comment]
Apologies, I had updated to the latest release, but hadn't restarted yet.

I can confirm that the behaviour/problem is still exhibited with the latest release.

```
Distribution: Arch Linux
GNOME Shell: 46.3.1
Display server: Wayland
PaperWM version: 46.13.2
Enabled extensions:
- system-monitor@gnome-shell-extensions.gcampax.github.com
- appindicatorsupport@rgcjonas.gmail.com
- paperwm@paperwm.github.com
- caffeine@patapon.info
```

Here is a screenshot showing the same behaviour under 46.13.2: ![Screenshot from 2024-07-18 17-59-21](https://github.com/user-attachments/assets/be3b9edb-19b7-4328-a1c2-b4005a52e135)

Regarding key bindings, I haven't changed those, so they should all be default still. So for take that's `Super+t` per default. And about the overlay, sometimes it's shown, others it doesn't. That doesn't seem to have any bearing on whether the bug is triggered or not though.

**Comment by jtaala** [comment]
hmm.  Is that a full screenshot?  Can I confirm you're seeing the floating tooltip?

![image](https://github.com/user-attachments/assets/44d9dd4d-fbf7-4cee-ae7d-93410e6a6a32)

I haven't been able to reproduce this at all (and have never had this happen on my systems).

Are you able to run (from a terminal):

```
journalctl -f /bin/gnome-shell
```

and reproduce the issue, and then paste the output above?

Thanks!

**Comment by jtaala** [comment]
I'll put a tag up to see if others have encountered this.

@Lythenas @Thesola10 @valpackett - have you ever seen this? (or can you reproduce?).

**Comment by jtaala** [comment]
> sometimes it's shown, others it doesn't

That's interesting - it should always show.  If it doesn't, it's symptom that something is wrong.  Hmmm, I'm guessing there's some exception on your setup that's stops the overlay showing (or it hits an issue before getting to there) and likely exits the function which puts it in a stuck state.

**Comment by jtaala** [comment]
Lastly to rule out an extension conflict - could you try disabling all other extensions (just having PaperWM enabled), log out, log back in, and try reproduce?

Thanks.

(Although your extensions don't look like an issue, and you only have a few...).  Still if we can rule that out that would get us a bit closer to the cause.

**Comment by cewood** [comment]
I've just verified that the problem still persists with all other extensions disabled, both before and after the reboot.
![Screenshot from 2024-07-19 11-32-57](https://github.com/user-attachments/assets/39033aa9-a63d-4cd0-bc0b-038464322bf5)

> hmm. Is that a full screenshot? Can I confirm you're seeing the floating tooltip?

Yes, all the screenshots I've been providing are full screenshots. When the problem is triggered, it appears the tooltip doesn't appear. And from what I've observed thus far, subsequent attempts to take windows, will show the tooltip correctly, except if the problem is triggered again.

Is there some logging I can enable to capture some info about this issue and hopefully help shed some more light on what's happening?

**Comment by jtaala** [comment]
Yep, see below:

> Are you able to run (from a terminal):
> 
> ```
> journalctl -f /bin/gnome-shell
> ```
> 
> and reproduce the issue, and then paste the output above?
> 
> Thanks!

**Comment by jtaala** [comment]
Bascially, run that command and gnome will start logging.  Try to reproduce and hopefully we should see the issue in the logging.

**Comment by cewood** [comment]
Here's the log output, I was able to trigger the problem twice during this period. The `meta_window_set_stack_position_no_sync` messages seemed to directly correlate to when the problem was triggered both times.

```
bohr% journalctl -f /bin/gnome-shell
Jul 19 13:15:29 bohr gnome-shell[1491]: meta_window_is_focusable: assertion '!window->unmanaging' failed
Jul 19 13:16:21 bohr gnome-shell[1491]: meta_window_is_focusable: assertion '!window->unmanaging' failed

Jul 19 13:34:44 bohr gnome-shell[1491]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Jul 19 13:34:45 bohr gnome-shell[1491]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed

Jul 19 13:35:08 bohr gnome-shell[1491]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
Jul 19 13:35:08 bohr gnome-shell[1491]: meta_window_set_stack_position_no_sync: assertion 'window->stack_position >= 0' failed
```

It's not always the case, but I found I can more reliably produce this problem by taking some windows, clicking the mouse, the windows become untaken, then take some windows again, then the windows, more often than not, become stuck once more, and no tooltip is shown. The missing tooltip seems to be a precursor/indication that the problem is about to occur.

**Comment by jtaala** [comment]
> It's not always the case, but I found I can more reliably produce this problem by taking some windows, clicking the mouse, the windows become untaken, then take some windows again, then the windows, more often than not, become stuck once more, and no tooltip is shown.

Nice one!  Yep, I can reproduce this with the above (clicking the mouse, try again, click mouse try again ==> issue occurs).  Thanks @cewood, given I can now reproduce, I should be able isolate and fix.

**Comment by jtaala** [comment]
Found the issue.  Keyboard grabstate was not released properly if exited "taking" windows by mouse click.

**Comment by jtaala** [comment]
Will do up a release shortly.

**Comment by jtaala** [comment]
The fixed version is `v46.13.5`.  It'll need to be approved on extensions.gnome.org (which sometimes takes a couple of hours).

**Comment by cewood** [comment]
Can confirm I'm unable to recreate the issue with `v46.13.5`. :tada: 
Thanks for the help and quick fix. :pray:
